streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/TaskMovementTest.java

```diff
@@ -35,262 +44,161 @@
 import static org.apache.kafka.streams.processor.internals.assignment.AssignmentTestUtils.UUID_3;
 import static org.apache.kafka.streams.processor.internals.assignment.AssignmentTestUtils.getClientStatesMap;
+import static org.apache.kafka.streams.processor.internals.assignment.AssignmentTestUtils.hasProperty;
 import static org.apache.kafka.streams.processor.internals.assignment.TaskMovement.assignTaskMovements;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.equalTo;
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.assertFalse;
-
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-import java.util.SortedSet;
-import java.util.UUID;
-import java.util.stream.Collectors;
-import org.apache.kafka.streams.processor.TaskId;
-import org.junit.Test;
+import static org.hamcrest.Matchers.is;
 
 public class TaskMovementTest {
-    private final ClientState client1 = new ClientState(1);
-    private final ClientState client2 = new ClientState(1);
-    private final ClientState client3 = new ClientState(1);
-
-    private final Map<UUID, ClientState> clientStates = getClientStatesMap(client1, client2, client3);
-
-    private final Map<UUID, List<TaskId>> emptyWarmupAssignment = mkMap(
-        mkEntry(UUID_1, EMPTY_TASK_LIST),
-        mkEntry(UUID_2, EMPTY_TASK_LIST),
-        mkEntry(UUID_3, EMPTY_TASK_LIST)
-    );
-
     @Test
     public void shouldAssignTasksToClientsAndReturnFalseWhenAllClientsCaughtUp() {
         final int maxWarmupReplicas = Integer.MAX_VALUE;
         final Set<TaskId> allTasks = mkSet(TASK_0_0, TASK_0_1, TASK_0_2, TASK_1_0, TASK_1_1, TASK_1_2);
 
-        final Map<UUID, List<TaskId>> balancedAssignment = mkMap(
-            mkEntry(UUID_1, asList(TASK_0_0, TASK_1_0)),
-            mkEntry(UUID_2, asList(TASK_0_1, TASK_1_1)),
-            mkEntry(UUID_3, asList(TASK_0_2, TASK_1_2))
-        );
-
         final Map<TaskId, SortedSet<UUID>> tasksToCaughtUpClients = new HashMap<>();
         for (final TaskId task : allTasks) {
             tasksToCaughtUpClients.put(task, mkSortedSet(UUID_1, UUID_2, UUID_3));
         }
-        
-        assertFalse(
+
+        final ClientState client1 = getClientStateWithActiveAssignment(asList(TASK_0_0, TASK_1_0));
+        final ClientState client2 = getClientStateWithActiveAssignment(asList(TASK_0_1, TASK_1_1));
+        final ClientState client3 = getClientStateWithActiveAssignment(asList(TASK_0_2, TASK_1_2));
+
+        assertThat(
             assignTaskMovements(
-                balancedAssignment,
                 tasksToCaughtUpClients,
-                clientStates,
-                getMapWithNumStandbys(allTasks, 1),
-                maxWarmupReplicas)
+                getClientStatesMap(client1, client2, client3),
+                maxWarmupReplicas),
+            is(false)
```

Comment:
The interface of assignTaskMovements has changed a little, since it no longer gets the assignments in a map and then applies them to the clientStates. Now, it just gets pre-assigned clientStates and mutates it. I think I've mostly faithfully reproduced the tests, but there were a few that I couldn't quite follow what they were previously doing.